src/host.cpp

```diff
@@ -284,101 +256,98 @@ namespace rhost {
         }
 
         void create_blob(const message& msg) {
-            assert(msg.name == "?CreateBlob");
+            assert(!strcmp(msg.name(), "?CreateBlob"));
+            blob_id id = create_blob(msg.blob());
+            respond_to_message(msg, static_cast<double>(id));
+        }
 
-            if (msg.blob.size() != msg.blob_count) {
-                fatal_error("Incomplete blob creation message %s.", msg.id.c_str());
-            }
+        blob_id create_blob(blob&& blob) {
+            std::lock_guard<std::mutex> lock(blobs_mutex);
+            blob_id id = ++next_blob_id;
 
-            {
-                std::lock_guard<std::mutex> lock(received_blobs_mutex);
-                int blob_id = ++next_blob_id;
-                received_blobs[blob_id] = msg.blob;
-                respond_to_message(msg, static_cast<double>(blob_id));
+            // Check that it never overflows double mantissa, and provide immediate diagnostics if it happens.
+            if (id != blob_id(double(id))) {
```

Comment:
The problem here is that we're casting it to `double` for round-tripping (because neither JSON nor R have 64-bit integer type), and `double` can only preserve 52 bits of value out of 64 (the rest are exponent and sign). The other problem is that when it happens, it's silent - you just lose precision. So I figured I'd add a check for that, just in case. Still highly unlikely that someone would create more than 2^52 blobs, so this is really more of a code-as-documentation.